clippy_lints/src/methods/path_join_correction.rs

```diff
@@ -0,0 +1,30 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, ty::is_type_diagnostic_item};
+use rustc_ast::ast::LitKind;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_span::{symbol::sym::Path, Span};
+
+use super::PATH_JOIN_CORRECTION;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>, join_arg: &'tcx Expr<'tcx>, span: Span) {
+    let ty = cx.typeck_results().expr_ty(expr);
+    if_chain!(
+     if is_type_diagnostic_item(cx, ty, Path);
+     let applicability = Applicability::MachineApplicable;
```

Comment:
No need to bind the applicability to a variable, don't worry :)
(Personal taste here, it's still fine)